src/libcrun/linux.c

```diff
@@ -559,15 +559,58 @@ format_mount_mapping (uint32_t container_id, uint32_t host_id, uint32_t size, si
   return format_mount_mappings (mappings, 1, written, direct);
 }
 
+static bool
+has_same_mappings (runtime_spec_schema_config_schema *def, runtime_spec_schema_defs_mount *mnt)
+{
+  size_t s;
+
+  if (def->linux == NULL)
+    return mnt->uid_mappings_len == 0 && mnt->gid_mappings_len == 0;
+
+  if (mnt->uid_mappings_len != def->linux->uid_mappings_len)
+    return false;
+
+  if (mnt->gid_mappings_len != def->linux->gid_mappings_len)
+    return false;
+
+  /* Compare host id and container id.  */
+
+  for (s = 0; s < mnt->uid_mappings_len; s++)
+    {
+      if (mnt->uid_mappings[s]->host_id != def->linux->uid_mappings[s]->container_id)
+        return false;
+      if (mnt->uid_mappings[s]->container_id != def->linux->uid_mappings[s]->host_id)
+        return false;
+      if (mnt->uid_mappings[s]->size != def->linux->uid_mappings[s]->size)
+        return false;
+    }
+
+  for (s = 0; s < mnt->gid_mappings_len; s++)
+    {
+      if (mnt->gid_mappings[s]->host_id != def->linux->gid_mappings[s]->container_id)
+        return false;
+      if (mnt->gid_mappings[s]->container_id != def->linux->gid_mappings[s]->host_id)
+        return false;
+      if (mnt->gid_mappings[s]->size != def->linux->gid_mappings[s]->size)
+        return false;
+    }
+
+  return true;
+}
+
 static pid_t
-create_userns_for_idmapped_mount (runtime_spec_schema_config_schema *def, runtime_spec_schema_defs_mount *mnt,
-                                  const char *options, libcrun_error_t *err)
+maybe_create_userns_for_idmapped_mount (runtime_spec_schema_config_schema *def,
+                                        runtime_spec_schema_defs_mount *mnt,
+                                        const char *options, pid_t *pid_out,
+                                        libcrun_error_t *err)
 {
-  cleanup_free char *dup_options = xstrdup (options);
-  char *option, *saveptr = NULL;
   cleanup_pid pid_t pid = -1;
   char proc_file[64];
   pid_t xchg_pid;
```

Comment:
Following variable is not being used anymore therefore causing lint issues.

Reply:
thanks, fixed now